core/src/main/kotlin/rys/ajaxpetproject/repository/impl/UserReactiveMongoRepository.kt

```diff
@@ -0,0 +1,63 @@
+package rys.ajaxpetproject.repository.impl
+
+import org.bson.types.ObjectId
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate
+import org.springframework.data.mongodb.core.findById
+import org.springframework.data.mongodb.core.findOne
+import org.springframework.data.mongodb.core.FindAndModifyOptions
+import org.springframework.data.mongodb.core.findAndModify
+import org.springframework.data.mongodb.core.remove
+import org.springframework.data.mongodb.core.findAll
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.data.mongodb.core.query.Query
+import org.springframework.data.mongodb.core.query.Update
+import org.springframework.stereotype.Repository
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import rys.ajaxpetproject.model.MongoUser
+import rys.ajaxpetproject.repository.UserRepository
+
+@Repository
+class UserReactiveMongoRepository(private val mongoTemplate: ReactiveMongoTemplate) : UserRepository {
+    override fun findById(id: ObjectId): Mono<MongoUser> {
+        val query = Query.query(Criteria.where("id").`is`(id))
+        return mongoTemplate.findById<MongoUser>(query)
+    }
+
+    override fun findByName(name: String): Mono<MongoUser> {
+        val query = Query.query(Criteria.where("userName").`is`(name))
+        return mongoTemplate.findOne<MongoUser>(query)
+    }
+
+    override fun save(user: MongoUser): Mono<MongoUser> = mongoTemplate.save(user)
+
+    override fun deleteAll(): Mono<Unit> {
+        return mongoTemplate.remove<MongoUser>(Query())
+            .doOnSuccess {  }
+            .thenReturn(Unit)
+    }
+
+    override fun update(id: ObjectId, user: MongoUser): Mono<MongoUser> {
+        val query = Query.query(Criteria.where("id").`is`(id))
+        val updatedUser = user.copy(id = id)
```

Comment:
I'm not sure why you need to copy the user here since it is not used. userName and password you can take from original user